### DISCUSSION TITLE
Vercel Serverless error.

### DISCUSSION THREAD

**Discussion by andregosling** [vercel/community#902] [Help] [answered]
Hi! I'm creating a url-shortener with Vercel serverless, and i got an error in some use-cases, that breaks the expected behavior (the final redirect). My flow is: The user access the shorted url (E.G. shorted.com/:token), then the service make a post to my backoffice to count the view of this endpoint, and make the redirect to my main domain (domain.com/:token). Note: I'm not using await on this request, with the intention to don't lose any speed at this redirect. 

The problem: For some times this approach fails, like when my backoffice got a delay more than usually to response, and the function crashes with this error bellow: 
<img width="1320" alt="image" src="https://user-images.githubusercontent.com/38388768/192607489-0ee340bb-9a10-4fd3-bb72-c762ed801b91.png">

I guess await the response of my backoffice will solve the problem, but also will delay my redirect. So, is there any better way to make this?

**Comment by amyegan** [comment, marked as answer]
Hi @andregosling! I think your guess is correct. You will need to `await` the response so the data is available to return before the request is complete
